### PR TITLE
1166: Added 500MB size limit to getUploadPolicy

### PR DESCRIPTION
**Pull request by sutt0n**
ustaxcourt/ef-cms#6794

### CODE REVIEW

**Review by kkoskelin — APPROVED** [review]
Sure.
@ericsorenson had a recent PR in which he was able to export the maximum file-size.  I'm wondering if we can use the same constant here, so that if(when) the max size changes it's easier? :)

File: shared/src/business/entities/Petition.js [hunk 1]
```diff
@@ -36,6 +40,7 @@ Petition.errorToMessageMap = {
   ownershipDisclosureFile: 'Ownership Disclosure Statement is required.',
   partyType: 'Party Type is a required field.',
   petitionFile: 'The Petition file was not selected.',
+  petitionFileSize: `Your file size is too big. The maximum file size is ${MAX_FILE_SIZE_MB}MB.`,
```
Comment (kkoskelin) [inline shared/src/business/entities/Petition.js]: You might want to structure multiple messages here similar to the above "irsNoticeDate" -- just in case they somehow select a zero-byte file?

**Review by kkoskelin — CHANGES_REQUESTED** [review]
Perhaps update the message map and add another test for a petitionFileSize of zero? Otherwise looks great.

File: shared/src/business/entities/Petition.js (hunk 1 above)
Comment (ericsorenson) [reply shared/src/business/entities/Petition.js]: This is a really good point.

**Review by ericsorenson — APPROVED** [review]

File: shared/src/business/entities/Petition.js (hunk 1 above)
Comment (sutt0n) [reply shared/src/business/entities/Petition.js]: We found out that you can only have one conditional object, and a default message:

```javascript
petitionFileSize: [
    {
      contains: 'must be less than or equal to',
      message: `Your Petition file size is too big. The maximum file size is ${MAX_FILE_SIZE_MB}MB.`,
    },
    'Your Petition file size is empty.',
  ],
```

We tried the below, and it seemed to error out in the `joiValidationDecorator`:

```javascript
petitionFileSize: [
{
      contains: 'must be larger than or equal to',
      message: `Your Petition file size is too small.`,
    },
    {
      contains: 'must be less than or equal to',
      message: `Your Petition file size is too big. The maximum file size is ${MAX_FILE_SIZE_MB}MB.`,
    },
    'Your Petition file size is non-existent.',
  ],
```

Dev Note: Found the `contains` verbiage in https://github.com/hapijs/joi/blob/master/lib/language.js

**Review by kkoskelin — APPROVED** [review]

**Review by matthopson — APPROVED** [review]
